code/modules/lost_crew/damages/_damages.dm

```diff
@@ -27,48 +29,76 @@
 	var/lore_death_time_max = 5 YEARS
 
 /// Generate and apply a possible character (species etc)
-/datum/corpse_damage_class/proc/apply_character(mob/living/carbon/human/fashion_corpse, list/protected_objects, list/datum/callback/on_revive_and_player_occupancy, list/body_data)
+/datum/corpse_damage_class/proc/apply_character(mob/living/carbon/human/fashion_corpse, list/protected_objects, list/recovered_items, list/datum/callback/on_revive_and_player_occupancy, list/body_data)
 	var/datum/corpse_character/character = pick_weight(possible_character_types)
 	character = new character()
-	character.apply_character(fashion_corpse, protected_objects, on_revive_and_player_occupancy)
+	character.apply_character(fashion_corpse, protected_objects, recovered_items, on_revive_and_player_occupancy)
 
 	var/datum/corpse_assignment/assignment = pick_weight(possible_character_assignments)
-	if(assignment)
+	if(ispath(assignment))
 		assignment = new assignment()
 		assignment.apply_assignment(fashion_corpse, protected_objects, on_revive_and_player_occupancy)
+		body_data += assignment.type
+
+	var/datum/corpse_flavor/flavor = pick_weight(possible_flavor_types)
+	if(ispath(flavor))
+		flavor = new flavor()
+		flavor?.apply_flavor(fashion_corpse, protected_objects, on_revive_and_player_occupancy)
+		body_data += flavor.type
 
 	death_lore += assignment?.job_lore
 
 	body_data += character.type
-	body_data += assignment?.type
 
 /// Set up injuries
 /datum/corpse_damage_class/proc/apply_injuries(mob/living/carbon/human/victim, list/saved_objects, list/datum/callback/on_revive_and_player_occupancy, list/body_data)
-	var/datum/corpse_damage/cause_of_death/cause_of_death = pick_damage_type(possible_causes_of_death)
-	cause_of_death.apply_to_body(victim, rand(), saved_objects, on_revive_and_player_occupancy)
+	var/bonus_roll = prob(80) //do an extra turn for cause of death or post mortum effect
+	var/bonus_roll_used = FALSE
+	var/list/used_damage_types = list()
+
+	var/datum/corpse_damage/cause_of_death/cause_of_death
+	for(var/i in 0 to 2)
+		cause_of_death = pick_damage_type(possible_causes_of_death, used_damage_types, bonus_roll_used)
+
+		if(!cause_of_death) //we ran out of causes of death (impressive!)
+			break
+
+		cause_of_death.apply_to_body(victim, rand(), saved_objects, on_revive_and_player_occupancy)
+		body_data += cause_of_death.type
+
+		if(cause_of_death.no_bonus_roll || !bonus_roll || prob(70))
```

Comment:
is it intentional that this has a separate `prob(70)` check to not do a bonus roll, when it already accounts for `!bonus_roll` which states
```dm
var/bonus_roll = prob(80) //do an extra turn for cause of death or post mortum effect
```

Reply:
yes this is intended. if this fails it will instead pass the bonus roll to post_mortem, where you can then both lose your organs AND limbs

Reply:
it feels a bit unintuitive without documentation, is it intended to be a 70% chance for a post mortum effect and a 30% chance for an additional cause of death?

Reply:
56% for extra death cause and 24% for post mortem (as in PR). I'll add a comment